webui/src/pages/repositories/wizard/wizard.jsx

```diff
@@ -1,100 +1,107 @@
+    const isStepOptional = (stepIndex) => {
+        return steps[stepIndex].optional;
+    };
 
-    const onStepChange = () => {
-        setRefresh(!refresh);
-        onNextStep();
+    const isStepSkipped = (stepIndex) => {
+        return skipped.has(stepIndex);
     };
 
-    const setInstance = stepWizard => {
-        setState({...state, stepWizard: stepWizard});
+    const isStepCompleted = (stepIndex) => {
+        return completed.has(stepIndex);
     }
 
-    return (
-        <Container className='container'>
-            <div className='jumbotron'>
-                <Row className={'justify-content-center'}>
-                    <Col>
-                        <StepWizard
-                            onStepChange={onStepChange}
-                            transitions={transitions}
-                            nav={showProgressBar && <WizardNav />}
-                            instance={setInstance}
-                            isHashEnabled={hashEnabled}
-                        >
-                            {children}
-                        </StepWizard>
-                    </Col>
-                </Row>
-            </div>
+    const handleNext = () => {
+        if (activeStep === steps.length - 1) {
+            onDone();
+        }
+        else {
+            let newSkipped = skipped;
+            if (isStepSkipped(activeStep)) {
+                newSkipped = new Set(newSkipped.values());
+                newSkipped.delete(activeStep);
+            }
+            setActiveStep((prevActiveStep) => prevActiveStep + 1);
+            setSkipped(newSkipped);
+        }
+    };
 
-            {<WizardController
-                stepWizard={state.stepWizard}
-                canProceed={canProceed}
-                skipButton={showSkipButton}
-                onComplete={onComplete}/>}
+    const handleBack = () => {
+        setActiveStep((prevActiveStep) => prevActiveStep - 1);
+    };
 
-        </Container>
-    );
-};
+    const handleSkip = () => {
+        if (!isStepOptional(activeStep)) {
+            throw new Error("You can't skip a step that isn't optional.");
+        }
+        if (activeStep === steps.length - 1) {
+            onDone();
+        }
+        else {
+            setActiveStep((prevActiveStep) => prevActiveStep + 1);
+            setSkipped((prevSkipped) => {
+                const newSkipped = new Set(prevSkipped.values());
+                newSkipped.add(activeStep);
+                return newSkipped;
+            });
+        }
+    };
 
-const WizardNav = ({totalSteps, currentStep}) => {
     return (
-        <ProgressBar className={'wizard-progress-bar'} striped max={totalSteps} min={1} now={currentStep} />
-    );
-}
+        <Box className={'jumbotron'}>
+            <Stepper activeStep={activeStep} alternativeLabel>
+                {steps.map((step, index) => {
+                    const stepProps = {};
+                    const labelProps = {};
```

Comment:
Opinion: these variables make the code harder to follow. I would prefer to see these props added inline below - see pseudo code example.